contracts/Wrapper.sol

```diff
@@ -32,53 +37,69 @@ contract Wrapper is PartialCommonOwnership721 {
     )
   {}
 
+  /// @notice helper function to get wrapperTokenId: hash the original NFT
+  ///         contract address and its respective local tokenId
+  /// @param assetContract Address the contract defining the token
+  /// @param tokenId Token Id to be wrapped
+  function createWrappedTokenId(
+    address assetContract, 
+    uint256 tokenId
+  ) public returns (uint256) {
+    return uint256(uint32(bytes4(keccak256(abi.encode(assetContract, tokenId)))));
+  }
+
+  /// @notice
+  /// @param operator Address of contract TODO
+  /// @param from Address the owner of the token
+  /// @param tokenId Token Id to be wrapped
+  /// @param _data extra data - unused
+  function onERC721Received(
```

Comment:
Safe Transferring the token to this contract requires subsequent method calls in order to set the price, beneficiary, tax rate and tax period.  I believe that it's better to have a single interface for depositing tokens into the PCO contract which allows the sender to set all of the relevant parameters in a single transaction.

Reply:
I fully agree with you that all parameters should preferably be set in a single atomic operation. My intention is that `acquire` is called by the client for this purpose.. 

I added a check in `onERC721Received` to make sure the tokenMap entry actually exists, so it can only be called as a result of calling `acquire`.

The advantage of this approach is that the owner of the wrapped token will be the same as the original token's owner by design.

If, however, the new owner would be passed as an argument via `acquire`, it could potentially be different from the original token's owner. In other words: someone could be changing ownership of the asset in the acquisition process, which is not a "feature" I think we should enable. The stricter the interface, the better in smart contract land, I think. But maybe I'm overthinking this, so happy to hear your thoughts :)

Reply:
Ah, that makes sense.